storage/innobase/row/row0import.cc

```diff
@@ -3075,6 +3178,151 @@ static size_t get_buf_size()
+      && cfg.m_n_indexes != UT_LIST_GET_LEN(table->indexes) + 1)
+    return nullptr;
+  dict_table_t *new_table= dict_table_t::create(
+    {table->name.m_name, strlen(table->name.m_name)},
+    table->space, table->n_t_cols - (DATA_N_SYS_COLS - 1),
+    table->n_v_cols, table->flags,
+    table->flags2);
+
+  new_table->id= table->id;
+  new_table->space_id= table->space_id;
+  size_t col_name_len= 0;
+  /* Copy columns from old table to new fts table */
+  for (ulint new_i= 0;
+       new_i < ulint(new_table->n_cols - (DATA_N_SYS_COLS + 1));
+       new_i++)
+  {
+    dict_mem_table_add_col(new_table, new_table->heap,
+                           &table->col_names[col_name_len],
+			   table->cols[new_i].mtype,
+			   table->cols[new_i].prtype,
+			   table->cols[new_i].len);
+    col_name_len+= strlen(&table->col_names[col_name_len]) + 1;
+  }
+
+  unsigned fts_col_ind= unsigned(table->n_cols - DATA_N_SYS_COLS);
+  fts_add_doc_id_column(new_table, new_table->heap);
+  new_table->cols[fts_col_ind].ind=
+    fts_col_ind & dict_index_t::MAX_N_FIELDS;
+  new_table->cols[fts_col_ind].ord_part= 1;
+  dict_table_add_system_columns(new_table, new_table->heap);
+
+  col_name_len= 0;
+  for (ulint new_i= 0; new_i < new_table->n_v_cols; new_i++)
+  {
+    dict_col_t old_vcol= table->v_cols[new_i].m_col;
+    ulint pos= old_vcol.ind;
+    if (old_vcol.ind <= fts_col_ind) pos++;
+    dict_mem_table_add_v_col(new_table, new_table->heap,
+                             &table->v_col_names[col_name_len],
+                             old_vcol.mtype, old_vcol.prtype,
+                             old_vcol.len, pos,
+                             table->v_cols[new_i].num_base);
+    for (ulint i= 0; i < table->v_cols[new_i].num_base; i++)
+    {
+      dict_col_t *base_col= dict_table_get_nth_col(
+        new_table, table->v_cols[new_i].base_col[i]->ind);
+      new_table->v_cols[new_i].base_col[i]= base_col;
+    }
+    col_name_len+= strlen(&table->v_col_names[col_name_len]) + 1;
+  }
+
+  /* Copy indexes from old table to new table */
+  for (dict_index_t *old_index= UT_LIST_GET_FIRST(table->indexes);
+       old_index;)
+  {
+    bool is_clustered= (old_index->type & DICT_CLUSTERED);
+    dict_index_t *new_index= dict_mem_index_create(
+      new_table, old_index->name, old_index->type,
+      is_clustered ? old_index->n_fields + 1 : old_index->n_fields);
+
+    new_index->id= old_index->id;
+    new_index->n_uniq= old_index->n_uniq;
+    new_index->type= old_index->type;
+    new_index->cached= 1;
+    /* Copy all fields from old index to new index */
+    for (ulint new_i= 0; new_i < new_index->n_fields; new_i++)
+    {
+      if (is_clustered && new_i == old_index->n_fields)
+      {
+        /* Add fts doc id in clustered index */
+        dict_index_add_col(
+          new_index, new_table, &table->cols[fts_col_ind], 0);
+        new_index->fields[new_i].fixed_len= sizeof(doc_id_t);
+        continue;
+      }
```

Comment:
Can we move the `if` body after the body of the `for` loop? It would seem to execute at the end of the loop.